src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java

```diff
@@ -0,0 +1,99 @@
+import org.springframework.security.oauth2.jwt.NimbusReactiveJwtDecoder.SecretKeyReactiveJwtDecoderBuilder;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+
+@Api(description = "Ride Request")
+@RequestMapping("/api/ride_request")
+@RestController
+
+public class RideController extends ApiController {
+
+    @Autowired
+    RideRepository rideRepository;
+
+    @ApiOperation(value = "List all rides, only users if not admin/driver")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER') || hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Ride> allRidesForUser() {
+        Iterable<Ride> rides;
+
+        if (getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_ADMIN")) ||
+            getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_DRIVER"))) {
+            rides = rideRepository.findAll();
+        } else {
+            rides = rideRepository.findAllByRiderId(getCurrentUser().getUser().getId());
+        }
+
+        return rides;
+    }
+
+    @ApiOperation(value = "Get a single ride by id, only if users if not admin/driver")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER') || hasRole('ROLE_USER')")
+    @GetMapping("")
+    public Ride getByIdForUser(
+            @ApiParam("id") @RequestParam Long id) {
+        Ride ride;
+        
+        if (getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_ADMIN")) ||
+            getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_DRIVER"))) {
+            ride = rideRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Ride.class, id));;
+        } else {
+            ride = rideRepository.findByIdAndRiderId(id, getCurrentUser().getUser().getId())
+                .orElseThrow(() -> new EntityNotFoundException(Ride.class, id));
+        }
+
+        return ride;
+    }
+
+    @ApiOperation(value = "Create a new ride")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public Ride postRide(
+        @ApiParam("day") @RequestParam String day,
+        @ApiParam("startTime") @RequestParam String startTime,
+        @ApiParam("endTime") @RequestParam String endTime,
+        @ApiParam("pickupLocation") @RequestParam String pickupLocation,
+        @ApiParam("dropoffLocation") @RequestParam String dropoffLocation,
+        @ApiParam("room") @RequestParam String room,
+        @ApiParam("course") @RequestParam String course
```

Comment:
Please follow this example on improving API endpoint clarity 

https://github.com/ucsb-cs156-s23/proj-courses-staff/pull/4

Reply:
Done!

Uploading Screen Recording 2023-06-06 at 2.48.02 PM.mov…